src/dmd/expressionsem.d

```diff
@@ -3588,6 +3589,32 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
 
         result = Expression.combine(argprefix, exp);
+
+        if (isSuper)
+        {
+            auto ad = sc.func ? sc.func.isThis() : null;
+            auto cd = ad ? ad.isClassDeclaration() : null;
+            if (cd && cd.classKind == ClassKind.cpp)
+            {
+                // if super is defined in C++, it sets the vtable pointer to the base class
+                // so we have to rewrite it, but still return 'this' from super() call:
+                // (auto tmp = super(), this.__vptr = __vtbl, tmp)
```

Comment:
This assignment cannot be interpreted at compile-time (unsupported cast), see https://forum.dlang.org/thread/ghusgzuqpcskhwzmlwbh@forum.dlang.org.

Reply:
How does CTFE assign vtables?

Reply:
Is this actually an issue? Any class with an opaque constructor (ie, is defined in C++) can't possibly CTFE...?

Reply:
Yep, if there's an external ctor, CTFE will fail anyway. If they're all in D, it used to work before, so it's clearly a regression, and resetting the vptr can probably be safely skipped during CTFE (something like `__ctfe || (this.__vptr = __vtbl)`).

Reply:
=> #8533

Reply:
Yeah, if that logic is only performed when the base constructor is extern, then it should still be fine.